packages/blade/src/components/Table/_decisions/decisions.md

```diff
@@ -0,0 +1,663 @@
+
+<img src="./table-thumbnail.png" width="760" alt="Table thumbnail" />
+
+- [Features](#features)
+- [Out of scope](#out-of-scope)
+- [Anatomy](#anatomy)
+- [API](#api)
+  - [Example Usage](#example-usage)
+  - [Detailed API](#detailed-api)
+    - [Table](#table)
+      - [`TableData`](#tabledata)
+      - [`SortFunctionsType`](#sortfunctionstype)
+      - [`SortChangeEvent`](#sortchangeevent)
+      - [`SelectionChangeEvent`](#selectionchangeevent)
+    - [TableToolbar](#tabletoolbar)
+    - [TableToolbarActions](#tabletoolbaractions)
+    - [TableHeader](#tableheader)
+    - [TableHeaderRow](#tableheaderrow)
+    - [TableHeaderCell](#tableheadercell)
+    - [TableBody](#tablebody)
+    - [TableRow](#tablerow)
+    - [TableCell](#tablecell)
+    - [TableFooter](#tablefooter)
+    - [TableFooterRow](#tablefooterrow)
+    - [TableFooterCell](#tablefootercell)
+    - [TablePagination](#tablepagination)
+      - [`PaginationChangeEvent`](#paginationchangeevent)
+      - [`RowCountPickerChangeEvent`](#rowcountpickerchangeevent)
+  - [API Design Decisions](#api-design-decisions)
+    - [1. Composable API](#1-composable-api)
+      - [Pros](#pros)
+      - [Cons](#cons)
+    - [2. Compact API](#2-compact-api)
+      - [Pros](#pros-1)
+      - [Cons](#cons-1)
+    - [Comparison](#comparison)
+    - [Decision](#decision)
+      - [1. Only Composable API](#1-only-composable-api)
+      - [2. Only Compact API](#2-only-compact-api)
+      - [3. Both Composable \& Compact API](#3-both-composable--compact-api)
+      - [Current API Usage Across Razorpay Projects](#current-api-usage-across-razorpay-projects)
+      - [Conclusion](#conclusion)
+- [Table Library Evaluation](#table-library-evaluation)
+  - [Why a library?](#why-a-library)
+  - [What are we looking for in a library?](#what-are-we-looking-for-in-a-library)
+  - [Libraries evaluated](#libraries-evaluated)
+    - [Heavy sized library - AG Grid](#heavy-sized-library---ag-grid)
+    - [Medium sized library - Tanstack's React Table](#medium-sized-library---tanstacks-react-table)
+    - [Small sized library - React Table Library](#small-sized-library---react-table-library)
+    - [Comparison](#comparison-1)
+- [Accessibility](#accessibility)
+
+
+## Features
+- Column Sorting
+- Row Selection - Single & Multiple
+- Pagination
+- Bulk Actions Toolbar
+- Horizontally Scrollable
+- Sticky Columns
+- Sticky Header
+- Sticky Footer
+- Cell Density - Normal & Comfortable
+
+## Out of scope
+We don't have enough use-cases for the following features at Razorpay and hence scoped them out of our Table component. If you are aware of any of these features being used at Razorpay, please let us know by creating an issue - we will evaluate it and scope it for future releases.
+
+- Column Reordering
+- Column Resizing
+- Column Filtering
+- Search
```

Comment:
Almost all merchant dashboard tables have need of search. Same concern as above, do we want to keep search controlled by external input fields? How do we maintain design/UX consistency this way?